python/singa/converter.py

```diff
@@ -0,0 +1,146 @@
+        if not file:
+            raise self.ProcessException("ERROR (" + filepath + ")!")
+        # Merges an ASCII representation of a protocol message into a message.
+        text_format.Merge(str(file.read()), parser_object)
+        file.close()
+        return parser_object
+
+    def convert_engine(self, layer_conf, caffe_solver):
+        '''
+        Convert caffe engine into singa engine
+        return:
+            a singa engine string
+        '''
+        caffe_engine = ''
+        solver_mode = caffe_solver.solver_mode
+        singa_engine = ''
+
+        # if no 'engine' field in caffe proto, set engine to -1
+        if layer_conf.type == 'Convolution' or layer_conf.type == 4:
+            caffe_engine = layer_conf.convolution_param.engine
+        elif layer_conf.type == 'Pooling' or layer_conf.type == 17:
+            caffe_engine = layer_conf.pooling_param.engine
+        elif layer_conf.type == 'ReLU' or layer_conf.type == 18:
+            caffe_engine = layer_conf.relu_param.engine
+        elif layer_conf.type == 'Sigmoid' or layer_conf.type == 19:
+            caffe_engine = layer_conf.sigmoid_param.engine
+        elif layer_conf.type == 'TanH' or layer_conf.type == 23:
+            caffe_engine = layer_conf.tanh_param.engine
+        elif layer_conf.type == 'LRN' or layer_conf.type == 15:
+            caffe_engine = layer_conf.lrn_param.engine
+        elif layer_conf.type == 'Softmax' or layer_conf.type == 20:
+            caffe_engine = layer_conf.softmax_param.engine
+        elif layer_conf.type == 'InnerProduct' or layer_conf.type == 14:
+            caffe_engine = -1
+        elif layer_conf.type == 'Dropout' or layer_conf.type == 6:
+            caffe_engine = -1
+        elif layer_conf.type == 'Flatten' or layer_conf.type == 8:
+            caffe_engine = -1
+        else:
+            raise Exception('Unknown layer type: ' + layer_conf.type)
+
+        # caffe_engine: -1-no field;  0-DEFAULT; 1-CAFFE; 2-CUDNN
+        # solver_mode: 0-CPU; 1-GPU
+        if solver_mode == 1:
+            singa_engine = 'cudnn'
+        else:
+            if caffe_engine == 2:
+                raise Exception('engine and solver mode mismatch!')
+            else:
+                singa_engine = 'singacpp'
+
+        if ((layer_conf.type == 'InnerProduct' or layer_conf.type == 14) or \
+            (layer_conf.type == 'Flatten' or layer_conf.type == 8)) and \
+            singa_engine == 'cudnn':
+            singa_engine = 'singacuda'
+
+        return singa_engine
+
+
+    def create_net(self):
+        '''
+        Create singa net based on caffe proto files.
+            net_proto: caffe prototxt that describes net
+            solver_proto: caffe prototxt that describe solver
+            input_sample_shape: shape of input data tensor
+        return:
+            a FeedForwardNet object
+        '''
+        caffe_net = self.read_net_proto()
+        caffe_solver = self.read_solver_proto()
```

Comment:
Maybe better make solver optional. 
For the prediction phase, we only need the net proto.
If it is not available, then just set solver_model to cpu.